siac/main.go

```diff
@@ -1,101 +1,117 @@
 package main
 
 import (
+	"encoding/json"
+	"net/http"
 	"os"
+	"reflect"
 
 	"github.com/spf13/cobra"
 )
 
-func versioncmd(*cobra.Command, []string) {
-	println("Sia Client v0.1.0")
+const (
+	VERSION  = "0.2.0"
+	hostname = "http://localhost:9980"
+)
+
+// getAPI makes an API call and decodes the response.
+func getAPI(call string, obj interface{}) (err error) {
+	resp, err := http.Get(hostname + call)
+	if err != nil {
+		return
+	}
+	err = json.NewDecoder(resp.Body).Decode(obj)
+	if err != nil {
+		return
+	}
+	resp.Body.Close()
+	return
+}
+
+// callAPI makes an API call and discards the response.
+func callAPI(call string) (err error) {
+	resp, err := http.Get(hostname + call)
+	if err != nil {
+		return
+	}
+	resp.Body.Close()
+	return
+}
+
+// wrap wraps a generic command with a check that the command has been
+// passed the correct number of arguments. The command must take only strings
+// as arguments.
+func wrap(fn interface{}) func(*cobra.Command, []string) {
+	fnVal, fnType := reflect.ValueOf(fn), reflect.TypeOf(fn)
+	if fnType.Kind() != reflect.Func {
+		panic("wrapped function has wrong type signature")
+	}
+	for i := 0; i < fnType.NumIn(); i++ {
+		if fnType.In(i).Kind() != reflect.String {
+			panic("wrapped function has wrong type signature")
+		}
```

Comment:
still uneasy about the panics. A panic isn't terribly user friendly. Could we print something and then log a stack trace?